rsocket-ipc-graphql/src/main/java/io/rsocket/graphql/GraphQLServerRequestStream.java

```diff
@@ -0,0 +1,85 @@
+package io.rsocket.graphql;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import graphql.execution.ExecutionId;
+import graphql.execution.instrumentation.Instrumentation;
+import graphql.schema.GraphQLSchema;
+import io.netty.buffer.ByteBuf;
+import io.rsocket.ipc.Functions;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoaderRegistry;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@SuppressWarnings("unchecked")
+class GraphQLServerRequestStream implements Functions.RequestStream<GraphQLRequest, Object> {
+  private final DataLoaderRegistry registry;
+  private final Instrumentation instrumentation;
+  private final GraphQLSchema graphQLSchema;
+
+  GraphQLServerRequestStream(
+      DataLoaderRegistry registry, Instrumentation instrumentation, GraphQLSchema graphQLSchema) {
+    this.registry = registry;
+    this.instrumentation = instrumentation;
+    this.graphQLSchema = graphQLSchema;
+  }
+
+  @Override
+  public Flux<Object> apply(GraphQLRequest request, ByteBuf byteBuf) {
+    try {
+      ExecutionInput.Builder builder =
+          ExecutionInput.newExecutionInput()
+              .query(request.getQuery())
+              .operationName(request.getOperationName())
+              .variables(request.getVariables())
+              .context(byteBuf)
+              .executionId(ExecutionId.generate());
+
+      if (registry != null) {
+        builder.dataLoaderRegistry(registry);
+      }
+
+      ExecutionInput executionInput = builder.build();
+
+      CompletableFuture<ExecutionResult> result =
+          GraphQL.newGraphQL(graphQLSchema)
+              .instrumentation(instrumentation)
+              .build()
+              .executeAsync(executionInput);
+
+      return Mono.fromFuture(result)
+          .flatMapMany(
+              executionResult -> {
+                List<GraphQLError> errors = executionResult.getErrors();
+
+                if (!errors.isEmpty()) {
+                  return Mono.error(GraphQLErrorException.from(errors));
+                }
+
+                if (!executionResult.isDataPresent()) {
+                  return Mono.empty();
+                }
+
+                Object r = executionResult.getData();
+
+                if (r == null) {
+                  return Mono.error(new NullPointerException("result data was null"));
+                }
+
+                if (r instanceof Publisher) {
```

Comment:
Does ExecutionInput.Builder drive whether reactive streams support is enabled?

Reply:
From that I can tell the DataFetcher return whatever you'd like. This is how the examples worked that I saw - they would cast the result.